packages/back-end/src/controllers/experiments.ts

```diff
@@ -1016,41 +988,39 @@ export async function deleteExperimentPhase(
   const { id, phase } = req.params;
   const phaseIndex = parseInt(phase);
 
-  const exp = await getExperimentById(org.id, id);
+  const experiment = await getExperimentById(org.id, id);
   const changes: Changeset = {};
 
-  if (!exp) {
+  if (!experiment) {
     res.status(404).json({
       status: 404,
       message: "Experiment not found",
     });
     return;
   }
 
-  if (exp.organization !== org.id) {
+  if (experiment.organization !== org.id) {
     res.status(403).json({
       status: 403,
       message: "You do not have access to this experiment",
     });
     return;
   }
 
-  req.checkPermissions("createAnalyses", exp.project);
+  req.checkPermissions("createAnalyses", experiment.project);
 
-  if (phaseIndex < 0 || phaseIndex >= exp.phases?.length) {
+  if (phaseIndex < 0 || phaseIndex >= experiment.phases?.length) {
     throw new Error("Invalid phase id");
   }
 
-  const existing = cloneDeep(exp);
-
   // Remove phase from experiment and revert to draft if no more phases left
-  exp.phases.splice(phaseIndex, 1);
-  changes.phases = exp.phases;
+  experiment.phases.splice(phaseIndex, 1);
```

Comment:
`splice` mutates the original array, which is not what you want here.  Instead, you can just use `filter`.

```ts
// Remove an element from an array without mutating the original
changes.phases = experiment.phases.filter((phase, i) => i !== phaseIndex);
```